### ISSUE TITLE
Unable to install v0.9.3 or v0.9.4 APKs on Android.

### ISSUE THREAD

**Issue by diarmidmackenzie** [tripleblindmarket/covid-safe-paths#543]
# Submitting an Issue

When submitting an Issue against Privte Kit, please follow these guidelines to help us help you.

## Version

Android 9
Motorola g7 

## Steps to reproduce

Delete previous versions of Safe Paths (0.9.2)
Download v0.9.3 or v0.9.4 APK
Attempt installation.

v0.9.3 installation fails with message "App not installed"

v0.9.4 installation tiggers a pop-up: Blocked by Play Protect.

"COVID Safe Paths.  Play protect does not recognize this app developer.  Apps from unknown developers can sometimes be unsafe."

Sleect "INSTALL ANYWAY".   Fails same as v0.9.3 "App not installed".

## Expected Behavior

I expected the app to install successfully (as 0.9.2 did).

Having read release notes I see v0.9.3 was a special minimal version for app store approval, so not interested in testing that.

But I do want to test v0.9.4

**Comment by kenpugsley** [comment]
Please retry the updated apk on 0.9.4.  I think it should now be signed correctly to include older android versions.  Once I hear from you that it fixes the issue, I'll rebuild 0.9.3

**Comment by kenpugsley** [comment]
I've gone ahead and updated the apk on 0.9.3 as well.  Tested the updated 0.9.4 apk on a device with android 6 and seems to install OK for me now.  Let me know how it goes for you.

**Comment by prasad79** [comment]
**Android** OS: 8.0
**Device**: Samsung Galaxy A3

Installation works fine if you do a clean install. i.e. uninstall 0.9.3 and install the new version.
Locally saved data would be lost though.

### Logs when updating version 0.9.3 to 0.9.4:

> 2020-04-17 07:23:46.855 4318-4361/? D/PackageInstallerCompatVL: add PAI Info to Cache : org.pathcheck.covidsafepaths, null, null
> 2020-04-17 07:23:46.855 4318-4361/? I/Launcher.IconCache: cachePackageInstallInfo packageName = org.pathcheck.covidsafepaths, title = null, icon = false
> 2020-04-17 07:23:46.856 4318-4361/? D/AppsModel: updateSessionDisplayInfo org.pathcheck.covidsafepaths
> 2020-04-17 07:23:48.612 3142-3230/? D/PackageManager: START_PACKAGE_INSTALL: observer{256921227}
>               stagedDir{/data/app/vmdl1569801179.tmp}
>               stagedCid{null}
>               Request from{com.google.android.packageinstaller}
>               VerificationInfo{originatingUri=https://github.com/tripleblindmarket/covid-safe-paths/releases/download/v0.9.4/COVIDSafePaths_0.9.4_1.apk,referrer=https://github.com/tripleblindmarket/covid-safe-paths/releases/download/v0.9.4/COVIDSafePaths_0.9.4_1.apk,originatingUid=10123,installerUid=10024}
> 2020-04-17 07:23:48.770 3142-3204/? D/PackageManager: EAS IT Policy org.pathcheck.covidsafepaths isPackageSignedByPlatform = false
> 2020-04-17 07:23:48.906 3142-3204/? D/PackageManager: EAS IT Policy org.pathcheck.covidsafepaths isPackageSignedByPlatform = false
> 2020-04-17 07:23:48.916 15433-15878/? D/GOS:MainIntentService: onHandleIntent(). PACKAGE_CHANGED. changeType : 2, packageName : org.pathcheck.covidsafepaths
> 2020-04-17 07:23:48.917 15433-15878/? D/GOS:DataManager: onPackageInstallStarted(). packageName :  org.pathcheck.covidsafepaths
> 2020-04-17 07:23:48.917 15433-15878/? D/GOS:DataManager: addPkgDataFromServer(), packageName : org.pathcheck.covidsafepaths
> 2020-04-17 07:23:48.918 15433-15878/? D/GOS:PkgData: PkgData(), pkgName: org.pathcheck.covidsafepaths
> 2020-04-17 07:23:48.918 15433-15878/? D/GOS:PkgData: setCustomFeatureFlag, pkgName: org.pathcheck.covidsafepaths, old featureFlag: 346781844236309267, new featureFlag: 346781844236309267
> 2020-04-17 07:23:48.920 15433-15878/? D/GOS:PkgData: PkgData(), pkgName: org.pathcheck.covidsafepaths
> 2020-04-17 07:23:48.920 15433-15878/? D/GOS:PkgData: setCustomFeatureFlag, pkgName: org.pathcheck.covidsafepaths, old featureFlag: 346781844236309267, new featureFlag: 346781844236309267
> 2020-04-17 07:23:48.920 3142-3204/? I/MSTG:  org.pathcheck.covidsafepaths is installing..
> _**_2020-04-17 07:23:48.949 3142-3204/? W/PackageManager: Package org.pathcheck.covidsafepaths signatures do not match the previously installed version; ignoring!_**_

**Comment by diarmidmackenzie** [comment]
Strangely I still couldn't install the new v0.9.4 on my phone.

Everything previously installed had been uninstalled.  Still insall fails in the same way.  Seems like the failed installs that I did have somehow got my phone into a state where it can't install the App.

On Android 9, Moto G7 power.

However, I moved to a 2nd account on my phone.  (actually the primary account; why was I using a secondary account before?  long story I won't go into now, but it's a recent change).

Now I can install the aop fine.

So maybe the issue is that the App cannot be installed on a non-primary account on an Android phone.  My notes aren't good enough so I don't know if I ever installed v0.9.2 on the secondary account.  I bet if I tried now it would fail.

I'll clone off another issue for this mysterious issue.  For now the app is up & running for me.

**Comment by anigasan** [comment]
I had an issue earlier on Galaxy Note 8 (Android 9) and rebooted after deleting earlier releases. It installed without any issue.

**Comment by prasad79** [comment]
- did the signing keys change between 0.9.3 to 0.9.4 versions?
- On google play store select -> enable **App Signing by Google Play**
- Close this issue, as future releases should not have this issue once the signing keys are fixed or **App Signing by Google Play** is enabled

**Comment by summetj** [comment]
Closing issue.

**Comment by diarmidmackenzie** [comment]
It turned out the issues I was having installing the app (other than the signing ithing) were in fact #563
